app/utils/ckb_utils.rb

```diff
@@ -38,24 +38,40 @@ def self.generate_lock_script_from_cellbase(cellbase)
   end
 
   def self.generate_address(lock_script)
-    return unless use_default_lock_script?(lock_script)
-
-    type1_address(lock_script)
+    if use_default_lock_script?(lock_script)
+      short_payload_blake160_address(lock_script)
+    else
+      full_payload_address(lock_script)
+    end
   end
 
-  def self.type1_address(lock_script)
+  def self.short_payload_blake160_address(lock_script)
     blake160 = lock_script.args.first
     return if blake160.blank? || !CKB::Utils.valid_hex_string?(blake160)
 
     CKB::Address.new(blake160).generate
   end
 
+  def self.full_payload_address(lock_script)
+    code_hash = lock_script.code_hash
+    args = lock_script.args
+    format_type = lock_script.hash_type == "data" ? "0x02" : "0x04"
+    first_arg = args.first
+
+    return unless args.all? { |arg| CKB::Utils.valid_hex_string?(arg) }
+
+    CKB::Address.new(first_arg).generate_full_payload_address(format_type, code_hash, args)
```

Comment:
It seems `first_arg` is useless, is this an SDK implement problem?

Reply:
SDK default params is blake160

Reply:
I changed SDK

Reply:
thanks.